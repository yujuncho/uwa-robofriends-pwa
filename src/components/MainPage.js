import React, { Component } from "react";

import CardList from "./CardList";
import SearchBox from "./SearchBox";
import Scroll from "./Scroll";
import ErrorBoundry from "./ErrorBoundry";
import Header from "./Header";

import "./MainPage.css";

class MainPage extends Component {
  componentDidMount() {
    this.props.onRequestRobots();
  }

  filteredRobots(robots, searchField) {
    return robots.filter(robot => {
      return robot.name.toLowerCase().includes(searchField.toLowerCase());
    });
  }

  render() {
    const { robots, searchField, onSearchChange, isPending } = this.props;
    return (
      <div className="tc">
        <Header />
        <SearchBox searchChange={onSearchChange} />
        <Scroll>
          {isPending ? (
            <h1>Loading</h1>
          ) : (
            <ErrorBoundry>
              <CardList robots={this.filteredRobots(robots, searchField)} />
            </ErrorBoundry>
          )}
        </Scroll>
      </div>
    );
  }
}

export default MainPage;
