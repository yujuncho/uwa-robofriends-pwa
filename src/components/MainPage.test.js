import { shallow } from "enzyme";
import MainPage from "./MainPage";

describe("MainPage", () => {
  let wrapper;

  beforeEach(() => {
    const mockProps = {
      onRequestRobots: jest.fn,
      robots: [],
      searchField: "",
      isPending: false
    };

    wrapper = shallow(<MainPage {...mockProps} />);
  });

  it("expects to render CardList", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("expects to render loading", () => {
    const mockProps = {
      onRequestRobots: jest.fn,
      robots: [],
      searchField: "",
      isPending: true
    };
    const wrapper = shallow(<MainPage {...mockProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("filters matching robots", () => {
    const robots = [{ name: "hello" }];
    expect(wrapper.instance().filteredRobots(robots, "h")).toEqual(robots);
    expect(wrapper.instance().filteredRobots(robots, "hello")).toEqual(robots);
  });

  it("returns empty when there are no matches", () => {
    expect(wrapper.instance().filteredRobots([])).toEqual([]);
    expect(wrapper.instance().filteredRobots([], "")).toEqual([]);

    const robots = [{ name: "hello" }];
    expect(wrapper.instance().filteredRobots(robots, "helloa")).toEqual([]);
    expect(wrapper.instance().filteredRobots(robots, "a")).toEqual([]);
  });
});
