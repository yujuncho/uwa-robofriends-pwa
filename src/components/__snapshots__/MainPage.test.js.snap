// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPage expects to render CardList 1`] = `
<div
  className="tc"
>
  <Header />
  <SearchBox />
  <Scroll>
    <ErrorBoundary>
      <CardList
        robots={Array []}
      />
    </ErrorBoundary>
  </Scroll>
</div>
`;

exports[`MainPage expects to render loading 1`] = `
<div
  className="tc"
>
  <Header />
  <SearchBox />
  <Scroll>
    <h1>
      Loading
    </h1>
  </Scroll>
</div>
`;
